fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergTransaction.java

```diff
@@ -59,140 +66,114 @@ public void updateIcebergCommitData(List<TIcebergCommitData> commitDataList) {
         }
     }
 
-    public void beginInsert(String dbName, String tbName) {
-        Table icebergTable = ops.getCatalog().loadTable(TableIdentifier.of(dbName, tbName));
-        transaction = icebergTable.newTransaction();
+    public void pendingCommit(SimpleTableInfo tableInfo) {
+        this.tableInfo = tableInfo;
+        this.transaction = getNativeTable(tableInfo).newTransaction();
```

Comment:
The table in the cache cannot be used here, because the table here is not necessarily the same as the cached table.